### DISCUSSION TITLE
Usage of UIVerticalScrollBar

### DISCUSSION THREAD

**Discussion by yunusayvaci** [MyreMylar/pygame_gui#268] [Q&A]
Hi. I want to usage of vertical bar but ı couldn't find its events in pyagame_gui document. How can ı use this element events loop in pygame? Please help me.
Thanks

**Comment by MyreMylar** [comment]
Hello,

I haven't created any events specifically for the scrollbars yet. In my own elements that use the scrollbars (the text box, selection list, scrolling window) I've been doing something like this:

```
if self.scroll_bar is not None and self.scroll_bar.check_has_moved_recently():
            self.scroll_bar.start_percentage # scrollbar has moved so do something here with start_percentage
```

You can see the code for these elements in the github repository if you want to explore them.

Out of curiosity - what are you using the scrollbar for that isn't covered by the existing elements? Also what sort of events were you thinking of? Something like `VERTICAL_SCROLLBAR_MOVED` like the `UIHorizontalSlider`?

**Reply by yunusayvaci** [reply]
I try to do  a pdf reader in pygame for learning. I find your pygame_gui. I like it for easy usage. I use button, drop down menu. I display bmp image of a page in pygame surface. Page is bigger than my pygame screen. I think that   if ı use vertical scrollbar, ı display page's part that out of screen with it.I don't know is it wrong or right? Maybe ı am wrong but in conclusion I strive for learn. Thanks

**Reply by yunusayvaci** [reply]
That is right. I thought there is a event like UIHorizontalSlider events
